File: src/demo_cpp/src/person_node.cpp
#include"rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    std::string name_;
    int age_;
public:
    PersonNode(
        const std::string & node_name,
        const std::string & name,
        const int & age
    )
    :Node(node_name) //same as super()
    {
        this->name_ = name;
        this->age_  = age;

    }
    void love(const std::string &lover)
    {
        RCLCPP_INFO(this->get_logger(),"i am %s,%d years old, love %s ",
        this->name_.c_str(),
        this->age_,
        lover.c_str()
    );
    }

};

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<PersonNode>("person_node","xiaoli",18);
    RCLCPP_INFO(node->get_logger(),"hello cpp node!");
    node->love(" xiaodiii");
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(
            self,
            node_name_v:str,
            name_v:str,
            age_v:int,
            ) -> None:
        print('using for adding two factors')
        super().__init__(node_name_v)
        self.name = name_v
        self.age = age_v

    def love(self, lover_v:str) :
        """
        way : love
        
        """
        self.get_logger().info(f"{self.age}th,{self.name},love {lover_v}")


def main():
    rclpy.init()
    node = PersonNode('Li,xiaolihaha', 18)
    xiaodi = PersonNode('di,xiaodihahah', 18)
    node.love('xiaodi')
    xiaodi.love('xiaoli')
    rclpy.spin(node)
    rclpy.shutdown()
